packages/ui/src/2_molecules/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,178 @@
+import React, {
+  FC,
+  ReactNode,
+  cloneElement,
+  useCallback,
+  Children,
+  useRef,
+  useState,
+  ReactElement,
+  useEffect,
+  useMemo,
+} from 'react';
+
+import classNames from 'classnames';
+
+import { Portal } from '../../1_atoms/Portal/Portal';
+import { useOnClickOutside } from '../../hooks/useOnClickOutside';
+import { Nullable } from '../../types';
+import { noop } from '../../utils';
+import styles from './Tooltip.module.css';
+import {
+  TooltipElements,
+  TooltipPlacement,
+  TooltipTrigger,
+  TooltipEvents,
+} from './Tooltip.types';
+import { CLOSE_DELAY, getTooltipPosition } from './Tooltip.utils';
+
+type TooltipProps = {
+  content: ReactNode;
+  children: ReactNode;
+  className?: string;
+  tooltipClassName?: string;
+  dataLayoutId?: string;
+  placement?: TooltipPlacement;
+  onShow?: () => void;
+  onHide?: () => void;
+  disabled?: boolean;
+  trigger?: TooltipTrigger;
+};
+
+export const Tooltip: FC<TooltipProps> = ({
+  content,
+  children,
+  className,
+  tooltipClassName,
+  dataLayoutId,
+  placement = TooltipPlacement.TOP,
+  onShow,
+  onHide,
+  disabled = false,
+  trigger = TooltipTrigger.hover,
+}) => {
+  const tooltipRef = useRef<HTMLDivElement>(null);
+  const targetRef = useRef<HTMLElement>(null);
+  const [elements, setElements] = useState<Nullable<TooltipElements>>(null);
+  const [isVisible, setIsVisible] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+  const [shouldHide, setShouldHide] = useState(false);
+  const isHoveredRef = useRef(isHovered);
+  isHoveredRef.current = isHovered;
+
+  const getElements = useCallback(() => {
+    const target = targetRef.current?.getBoundingClientRect();
+    const tooltip = tooltipRef.current?.getBoundingClientRect();
+    if (target && tooltip) {
+      return { target, tooltip };
+    }
+    return null;
+  }, []);
+
+  const updateElements = useCallback(() => {
+    const elements = getElements();
+    setElements(elements);
+  }, [getElements, setElements]);
+
+  const tooltipPosition = useMemo(() => {
+    if (!elements) {
+      return;
+    }
+    return getTooltipPosition(elements, placement);
+  }, [elements, placement]);
+
+  const handleShow = useCallback(() => {
+    setIsVisible(prevValue => !prevValue);
+    if (isVisible) {
+      setShouldHide(true);
+    }
+  }, [setIsVisible, setShouldHide, isVisible]);
+
+  const handleHide = useCallback(() => {
+    setShouldHide(true);
+  }, [setShouldHide]);
+
+  const onMouseHover = useCallback(() => {
+    setIsHovered(prevValue => !prevValue);
+  }, [setIsHovered]);
+
+  const getElementProps = useCallback(() => {
+    const attributes = {
+      'data-layout-id': dataLayoutId,
```

Comment:
I'd create `DATA_ATTRIBUTE` constant in `types/index.tsx` so we don't need to hardcode `data-layout-id` every time we need to use it as string.

Reply:
@tiltom good suggestion, but let's also apply it to the other components after this PR is merged - added new story for this here: https://sovryn.atlassian.net/browse/SOV-861